### PR TITLE
Feat(BCE-41): View Different Types of Reports

### PR DESCRIPTION
**Pull request by cjayneb**
JIRA: [link to jira ticket](https://champlainsaintlambert.atlassian.net/jira/software/projects/BCE/boards/22?selectedIssue=BCE-41)
## Context:
This PR is to add the ability to see the types of reports available to the admin user so that they can perform actions on them later (like modifying the Recipients list). The only type of Report available right now is the monthly Employee Performance Report.

## Changes
- Added model for Report Types.
- Added model for Report Recipients.
- Added model to connect Recipients with Types => ReportTypeRecipientModel.
- Initialize Report Types with the Monthly Employee Performance Report.
- Initialize Report Recipients with the list of recipients given by the client.
- Initialize ReportTypeRecipient with list of recipients for the Monthly Employee Performance Report.
- Added backend functions to get all Report Types with each of their recipients.
- Added Report DAO for getting Report Types and Recipients.
- Made Reports page layout dynamic depending on screen size and user role.
- Added Reports management section on the Reports Page.
- Added Select box to select the type of Report
- Added disabled select box for frequency (will be enabled later when the features is available)
- Added scrollable list to see the selected Report Types' recipients.

## Before and After UI (Required for UI-impacting PRs)
BEFORE
image

AFTER
Reports Page for an admin user (Max width)
image

Reports Page for an admin user (normal width)
image

Report Page for an admin user (small width)
image

Reports page for an employee user
image

## Dev notes (Optional)
- Whoever is doing the Report Recipient List story, make sure to follow the established models.

### CODE REVIEW

File: client/src/pages/Reports.js [hunk 1]
```diff
@@ -36,24 +39,51 @@ const Reports = () => {
...
 
         Axios.defaults.withCredentials = true;
 
-        Axios.get(`${process.env.REACT_APP_API}/reports/chartReport`, { headers: header })
+        await Axios.get(`${process.env.REACT_APP_API}/reports/reportTypes`, { headers: header })
             .then((response) => {
-                setChartReports(response.data);
+                console.log(response)
+                if (response.data) {
+                    console.log(response.data)
```
Comment (NickEspitalier) [inline client/src/pages/Reports.js:73]: Yeah, if you don't need to use the console.log, remove them

File: server/data_access_layer/models/localdb/localdb.model.js [hunk 2]
```diff
@@ -189,21 +197,69 @@ module.exports = (localdb, Sequelize) => {
...
+            field: 'recipient_name',
+            type: Sequelize.STRING
+        },
+        email: {
+            field: 'recipient_email',
+            type: Sequelize.STRING
+        }
+    })
+
+    const ReportTypeRecipients = localdb.define("report_type_recipients", {
```
Comment (NickEspitalier) [inline server/data_access_layer/models/localdb/localdb.model.js:243]: You can put the constant whole statement in one line if there is no key value pair declared inside,

**Review by NickEspitalier — APPROVED** [review]
Looks good, you just have to change the formatting of your code a little bit. LGTM

File: server/data_access_layer/databases.js [hunk 3]
```diff
@@ -112,8 +115,90 @@ db.sync = async (database, options) => {
...
+      for (let i = 0; i < data.recipients.length; i++) {
+        reportTypeRecipients.push({
+          report_type_id: data.reportTypes.reportTypeId,
+          recipient_id: data.recipients[i].recipientId
+        });
+      }
+      await db['localdb'].reportTypeRecipients.bulkCreate(reportTypeRecipients);
+    })
     .catch((err) => {
-      console.log(err);
```
Comment (Funnyadd) [inline server/data_access_layer/databases.js:116]: Often, console logs have to be removed because they resolve unnecessary data.

**Review by Funnyadd — APPROVED** [review]
LGTM brother, great work!

**Pull request merged**